srcjs/input_binding_date.js

```diff
@@ -16,10 +16,17 @@ $.extend(dateInputBinding, {
   setValue: function(el, value) {
     var date = this._newDate(value);
     // If date is invalid, do nothing
+    // Note: R's NA is null here, and isNaN(null) is false
```

Comment:
It is now. :) See my change below at https://github.com/rstudio/shiny/pull/1299/files/3707cd813f1262410681a8f1f0b2f1186081ad73#diff-664d9420c5db5c8d072c778dcb0be113R155

Is that OK? I am a bit worried that I cannot unit test this stuff....